CODE_OF_CONDUCT.md

```diff
@@ -2,75 +2,134 @@
 
 ## Our Pledge
```

Comment:
do we need this file and our own version of it? I imagine this will be internally driven, not from the community (which is okay!)

Reply:
Since it's a public repo, I think we should leave it in. But you're right that I would be quite surprised if we got contributions from outside MongoDB. And if we did, I think we'd probably just ask them to submit them to Uber's version instead.